modules/presence/presentity.c

```diff
@@ -555,12 +555,19 @@ int update_presentity(struct sip_msg* msg, presentity_t* presentity, int* sent_r
 			/* wait to get our turn as order of handling pubishs
 			   (need to wait the ongoing published to terminate
 			   before starting */
+			int i = 0;
 			while (p && turn!=p->current_turn) {
 				lock_release(&pres_htable[hash_code].lock);
 				sleep_us(100);
 				lock_get(&pres_htable[hash_code].lock);
 				p = search_phtable_etag(&pres_uri, presentity->event->evp->parsed,
 					&presentity->etag, hash_code);
+				i ++;
+				if ((i >= 30) && p && (turn!=p->current_turn)) {
```

Comment:
What's so special about 30?  Should at least add a comment as to why but still...why?